### PR TITLE
BUGFIX: SETI-190 Zuul: report back to pull request when no jobs are run

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by michalvanco** [comment]
please rebase

File: tests/test_scheduler.py [hunk 1]
```diff
@@ -1982,6 +1982,26 @@ def test_no_job_project(self):
...
+        self.config.set('zuul', 'layout_config',
+                        'tests/fixtures/layout-no-job-report.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+        A = self.fake_gerrit.addFakeChange('org/no-jobs-project2',
+                                           'master', 'A')
+        self.fake_gerrit.addEvent(A.getPatchsetCreatedEvent(1))
+        self.waitUntilSettled()
+
+        # Change wasn't reported to
```
Comment (hrubi) [inline tests/test_scheduler.py]: Change _was_ reported to

File: tests/fixtures/layout-no-job-report.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      gerrit:
+        verified: 1
+    failure:
+      gerrit:
+        verified: -1
+
+projects:
+  - name: org/no-jobs-project2
+    check-always-report:
+      - project-testfile
```
Comment (hrubi) [inline tests/fixtures/layout-no-job-report.yaml]: This _will_ run. You want some filefilter to skip the job. Eg. a same one as is in tests/fixtures/layout.yaml for the project-testfile job.

File: tests/test_scheduler.py [hunk 3]
```diff
@@ -1982,6 +1982,26 @@ def test_no_job_project(self):
...
+        self.assertEqual(A.reported, True)
+
+        # Check queue is empty afterwards
+        check_pipeline = self.sched.layout.pipelines['check-always-report']
+        items = check_pipeline.getAllItems()
+        self.assertEqual(len(items), 0)
+        self.assertEqual(len(self.history), 1)
```
Comment (hrubi) [inline tests/test_scheduler.py]: You wanted to test that no job did run, but the pipeline still got reported. Here you are asserting that one job did run.

File: tests/test_scheduler.py [hunk 4]
```diff
@@ -1982,6 +1982,26 @@ def test_no_job_project(self):
...
+                        'tests/fixtures/layout-no-job-report.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+        A = self.fake_gerrit.addFakeChange('org/no-jobs-project2',
+                                           'master', 'A')
+        self.fake_gerrit.addEvent(A.getPatchsetCreatedEvent(1))
+        self.waitUntilSettled()
+
+        # Change wasn't reported to
+        self.assertEqual(A.reported, True)
```
Comment (hrubi) [inline tests/test_scheduler.py]: Assert the reported status too.

File: zuul/scheduler.py [hunk 5]
```diff
@@ -505,6 +505,9 @@ def _parseConfig(self, config_path, connections):
                     manager.event_filters += trigger.getEventFilters(
                         conf_pipeline['trigger'][trigger_name])
 
+            pipeline.report_empty = conf_pipeline.get('report-empty',
```
Comment (hrubi) [inline zuul/scheduler.py]: Define the `report_empty` attribute in `model.Pipeline`.

File: tests/base.py [hunk 6]
```diff
@@ -1271,6 +1271,7 @@ def setUp(self):
         self.init_repo("org/noop-project")
         self.init_repo("org/experimental-project")
         self.init_repo("org/no-jobs-project")
+        self.init_repo("org/no-jobs-project2")
```
Comment (hrubi) [inline tests/base.py]: You should be OK with any other repo. No need to add a new one and prolong the setup time.

File: tests/test_scheduler.py (hunk 3 above)
Comment (hrubi) [inline tests/test_scheduler.py]: Also separate the history assert with whitespace as this line no more logically belongs to "Check queue is empty afterwards". See `test_no_job_project` above.

File: tests/test_scheduler.py [hunk 7]
```diff
@@ -1982,6 +1982,28 @@ def test_no_job_project(self):
...
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+        A = self.fake_gerrit.addFakeChange('org/no-jobs-project',
+                                           'master', 'A')
+        self.fake_gerrit.addEvent(A.getPatchsetCreatedEvent(1))
+        self.waitUntilSettled()
+
+        # Change was reported to
+        self.assertEqual(A.reported, True)
+        self.assertEqual(A.data['status'], 'MERGED')
```
Comment (hrubi) [inline tests/test_scheduler.py]: Sorry, I meant to assert this:
```python
self.assertEqual('Build succeeded.\n\n', A.messages[0])
```

**Pull request merged**